src/EditorFeatures/Core/Implementation/NavigationBar/NavigationBarController_ModelComputation.cs

```diff
@@ -142,11 +140,15 @@ private void StartSelectedItemUpdateTask(int delay, bool updateUIWhenDone)
             if (updateUIWhenDone)
             {
                 asyncToken = _asyncListener.BeginAsyncOperation(GetType().Name + ".StartSelectedItemUpdateTask.UpdateUI");
-                _selectedItemInfoTask.SafeContinueWith(
-                    t => PushSelectedItemsToPresenter(t.Result),
+                _selectedItemInfoTask.SafeContinueWithFromAsync(
+                    async t =>
+                    {
+                        await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
❕ This needs to yield if called from the main thread in order to preserve the original behavior.